crates/egui/src/containers/menu.rs

```diff
@@ -446,7 +446,7 @@ impl SubMenu {
         let is_hovered = hover_pos.is_some_and(|pos| button_rect.contains(pos));
 
         // The clicked handler is there for accessibility (keyboard navigation)
-        if (!is_any_open && is_hovered) || button_response.clicked() {
+        if (!is_any_open && is_hovered && ui.is_enabled()) || button_response.clicked() {
```

Comment:
This looks like it will still open when clicked, enabled or not (though I guess there is another path preventing it from becoming `clicked` when disabled).

```suggestion
        let should_open = ui.is_enabled() && (button_response.clicked() || (is_hovered && !is_any_open));
        if should_open {
```